cmake/CheckTrezor.cmake

```diff
@@ -138,22 +116,38 @@ if(Protobuf_FOUND AND USE_DEVICE_TREZOR AND TREZOR_PYTHON)
 endif()
 
 # Try to build protobuf messages
-if(Protobuf_FOUND AND USE_DEVICE_TREZOR AND TREZOR_PYTHON)
-    set(ENV{PROTOBUF_INCLUDE_DIRS} "${Protobuf_INCLUDE_DIR}")
-    set(ENV{PROTOBUF_PROTOC_EXECUTABLE} "${Protobuf_PROTOC_EXECUTABLE}")
-    set(TREZOR_PROTOBUF_PARAMS "")
-    if (USE_DEVICE_TREZOR_DEBUG)
-        set(TREZOR_PROTOBUF_PARAMS "--debug")
-    endif()
-    
-    execute_process(COMMAND ${TREZOR_PYTHON} tools/build_protob.py ${TREZOR_PROTOBUF_PARAMS} WORKING_DIRECTORY ${CMAKE_CURRENT_LIST_DIR}/../src/device_trezor/trezor RESULT_VARIABLE RET OUTPUT_VARIABLE OUT ERROR_VARIABLE ERR)
+if(Protobuf_FOUND AND USE_DEVICE_TREZOR)
+    # .proto files to compile
+    set(_proto_files "messages.proto"
+                     "messages-common.proto"
+                     "messages-management.proto"
+                     "messages-monero.proto")
+    if (TREZOR_DEBUG)
+        list(APPEND _proto_files "messages-debug.proto")
+    endif ()
+
+    set(_proto_include_dir "${CMAKE_SOURCE_DIR}/external/trezor-common/protob")
+    set(_proto_files_absolute)
+    foreach(file IN LISTS _proto_files)
+        list(APPEND _proto_files_absolute "${_proto_include_dir}/${file}")
+    endforeach ()
+
+    set(_proto_out_dir "${CMAKE_SOURCE_DIR}/src/device_trezor/trezor/messages")
+    execute_process(COMMAND ${Protobuf_PROTOC_EXECUTABLE} --cpp_out "${_proto_out_dir}" "-I${_proto_include_dir}" ${_proto_files_absolute} RESULT_VARIABLE RET OUTPUT_VARIABLE OUT ERROR_VARIABLE ERR)
     if(RET)
-        trezor_fatal_msg("Trezor: protobuf messages could not be regenerated (err=${RET}, python ${PYTHON})."
-                "OUT: ${OUT}, ERR: ${ERR}."
-                "Please read src/device_trezor/trezor/tools/README.md")
+        trezor_fatal_msg("Trezor: protobuf messages could not be (re)generated (err=${RET}). OUT: ${OUT}, ERR: ${ERR}.")
+    endif()
+
+    if(FREEBSD)
+        # FreeBSD defines `minor` in usr/include/sys/types.h which conflicts with this file
+        file(READ "${_proto_out_dir}/messages-monero.pb.h" file_content)
+        string(REPLACE "// @@protoc_insertion_point(includes)"
```

Comment:
I suspect this aims to address the following bug:
https://github.com/protocolbuffers/protobuf/issues/4654

For tracking purposes, it would help to link the relevant github issue. Per the discussion on that issue, the bug may be fixed in more recent versions of protocol buffers, so it's worth checking in https://github.com/monero-project/monero/pull/9478.

Reply:
I tried these changes with #9478, but it doesn't make a difference. [This comment](https://github.com/protocolbuffers/protobuf/issues/4654#issuecomment-391457742) on the linked issue seems to suggest renaming the [problematic field](https://github.com/trezor/trezor-common/blob/master/protob/messages-monero.proto#L91) is the only solution. 

I have opened [this issue](https://github.com/trezor/trezor-firmware/issues/4460) on the Trezor firmware repo and force pushed a change to this PR that references it.

Reply:
The comment you referenced pre-dates the fix. I'm not certain that the latest protobuf releases really fixed the issue, but https://github.com/protocolbuffers/protobuf/issues/4654#issuecomment-612931531 suggests maybe they did.

Reply:
The fix is likely this [commit](https://github.com/protocolbuffers/protobuf/commit/416f90939d4de58fe1a4e2489120010313183291), which essentially adds the ifdef undef block after the `protoc_insertion_point` like we do here.

This was later [changed](https://github.com/protocolbuffers/protobuf/blob/a2a284004dbe5084c1ef1aa3b9e707827c46b5bc/src/google/protobuf/compiler/cpp/file.cc#L194-L196) to only do this for protobuf's own types.